### ISSUE TITLE
System issue - cannot download PDFs, IIIF images

### ISSUE THREAD

**Issue by Melissa37** [microsimulation/ijm-xml#60]
![Screenshot 2019-05-17 at 15 21 19](https://user-images.githubusercontent.com/6051876/57934388-8a879380-78b7-11e9-8c48-c44a1f1804f6.png)

Click on PDF, then replaces currently page with:

![Screenshot 2019-05-17 at 15 21 25](https://user-images.githubusercontent.com/6051876/57934393-8d828400-78b7-11e9-9c05-cb686714f730.png)

**Comment by gnott** [comment]
For me, PDFs loaded from the `http://localhost:8080/` URL when running Docker locally. I think to fix this will be in the demo site with some configuration, if you are viewing a live demo site?

**Comment by Melissa37** [comment]
Yes, I am using:
https://demo--ijm.elifesciences.org/articles

**Comment by giorgiosironi** [comment]
Probably a similar issue to https://github.com/microsimulation/demo/pull/10 but let's solve that first and we'll check where this other hardcoded URL comes from (if it's not already covered).

**Comment by gnott** [comment]
Not entirely sure where the download configuration is from yet, I am following it to see.

The IJM content is similar to this line from the example branch https://github.com/microsimulation/demo/blob/example/api/data/articles/04395.json#L164

```
"pdf": "http://web:8082/04395/elife-04395-fig1-v2.tif"
```

I followed that pattern when generating IJM JSON files.

**Comment by gnott** [comment]
E.g.

https://github.com/microsimulation/demo/blob/master/api/data/articles/00001.json#L15
```
    "pdf": "http://web:8082/00001/ijm-00001.pdf",
```

**Comment by giorgiosironi** [comment]
Uhm
http://demo--ijm.elifesciences.org/download/aHR0cDovL3dlYjo4MDgyLzAwMDUzL2lqbS0wMDA1My5wZGY=/ijm-00053.pdf?_hash=B5BO4Z7yWQiD1gIrsfInUViscVdKIJclwfKseD%2FFbKg%3D
works whereas the original link
http://localhost:8080/download/aHR0cDovL3dlYjo4MDgyLzAwMDUzL2lqbS0wMDA1My5wZGY=/ijm-00053.pdf?_hash=B5BO4Z7yWQiD1gIrsfInUViscVdKIJclwfKseD%2FFbKg%3D
doesn't as `localhost` can't resolve.
What should happen, I think, is that the hostname here resolves to the `journal` container and that proxies the request to the `web:8082` that you have used. `web:8082` only resolves from inside containers.

**Comment by giorgiosironi** [comment]
I think the problem is in the [proxying](https://github.com/elifesciences/microsimulation-demo-formula/blob/master/salt/microsimulation-demo/config/etc-nginx-sites-enabled-microsimulation-demo.conf#L8) that does not forward headers such as `Host` or `X-Forwarded-Host`.

**Comment by giorgiosironi** [comment]
Symfony ignores my attempts to set `Host` or `X-Forwarded-Host` with `proxy_set_header`.

**Comment by giorgiosironi** [comment]
As far as I can understand the settings in https://symfony.com/doc/3.4/console/request_context.html are only defaults so we can't go that route either.

**Comment by giorgiosironi** [comment]
Added `Request::setTrustedProxies([$_SERVER['REMOTE_ADDR']], Request::HEADER_X_FORWARDED_ALL);` which doesn't help either.

**Comment by giorgiosironi** [comment]
https://github.com/microsimulation/demo/pull/13
https://github.com/elifesciences/microsimulation-demo-formula/pull/4
seem to work for me, HTTPS pending as that can only be tested on AWS.

**Comment by giorgiosironi** [comment]
IIIF images have similar problems still, but the resolution needs to wait on the proxying fixes as they will need to be proxied as well to add HTTPS.

**Comment by giorgiosironi** [comment]
Finally fixed PDF links; images are a different matter.

**Comment by gnott** [comment]
Cool, thanks @giorgiosironi and others. It works for me on the demo site.

**Comment by giorgiosironi** [comment]
I suspect [`api_url_public` is unused](https://github.com/microsimulation/demo/search?q=api_url_public&unscoped_q=api_url_public)? We can allow public access without configuring it in journal.

**Comment by giorgiosironi** [comment]
Images fixed via more proxying.
<img src="https://pbs.twimg.com/media/D75tM-PWsAMO2A4.jpg:large" />

**Comment by thewilkybarkid** [comment]
Think this can be closed now.
